esrally/utils/git.py

```diff
@@ -48,6 +48,23 @@ def is_working_copy(src):
     return os.path.exists(src) and os.path.exists(os.path.join(src, ".git"))
 
 
+@probed
+def is_branch(src_dir, remote, identifier):
+    fetch(src_dir, remote=remote)
+    show_ref_cmd = f"git -C {src_dir} show-ref {identifier}"
+
+    # if we get an non-zero exit code, we know that the identifier is not a branch (local or remote)
+    if not process.exit_status_as_bool(lambda: process.run_subprocess_with_logging(show_ref_cmd)):
+        return False
+
+    # it's possible the identifier could be a tag, so we explicitly check that here
+    ref = process.run_subprocess_with_output(show_ref_cmd)
+    if "refs/tags" in ref[0]:
+        return False
```

Comment:
The double call here is a bit sad, but I agree that our current process abstractions enforce it. We could add a wrapper around subprocess.run(), but I think that's not needed as this is a fast local call anyway.

Reply:
++, I didn't want to add a new process function in this PR. I raised https://github.com/elastic/rally/issues/1595 to track.